src/mlpack/core/data/save_impl.hpp

```diff
@@ -106,57 +142,54 @@ bool Save(const std::string& filename,
     Timer::Stop("saving_data");
     return false;
   }
+  const bool isImageFormat = (opts.Format() == FileType::PNG ||
+      opts.Format() == FileType::JPG || opts.Format() == FileType::PNM ||
+      opts.Format() == FileType::BMP || opts.Format() == FileType::GIF ||
+      opts.Format() == FileType::PSD || opts.Format() == FileType::TGA ||
+      opts.Format() == FileType::PIC || opts.Format() == FileType::ImageType);
 
   // Try to save the file.
   Log::Info << "Saving " << opts.FileTypeToString() << " to '" << filename
       << "'." << std::endl;
-  if constexpr (IsArma<ObjectType>::value || IsSparseMat<ObjectType>::value)
+  if constexpr (isMatrixType)
   {
-    TextOptions txtOpts(std::move(opts));
-    if constexpr (IsSparseMat<ObjectType>::value)
-    {
-      success = SaveSparse(matrix, txtOpts, filename, stream);
-    }
-    else if constexpr (IsCol<ObjectType>::value)
+    if (isImageFormat)
     {
-      opts.NoTranspose() = true;
-      success = SaveDense(matrix, txtOpts, filename, stream);
-    }
-    else if constexpr (IsRow<ObjectType>::value)
-    {
-      opts.NoTranspose() = false;
-      success = SaveDense(matrix, txtOpts, filename, stream);
+      if constexpr (isSparseMatrixType)
+      {
+        return HandleError("Cannot load image data into a sparse matrix. "
+        "Please use dense matrix instead.", opts);
```

Comment:
Here too I would suggest a super-simple two-line way to make this work:

```
arma::Mat<typename ObjectType::elem_type> tmp =
    conv_to<arma::Mat<typename ObjectType::elem_type>>::from(matrix);
Save(filename, tmp, opts);
```

Or you can make it slightly more complex and call `SaveImage()` directly, either way is fine.